src/Microsoft.DotNet.Interactive/Parsing/PolyglotLexer.cs

```diff
@@ -207,9 +215,11 @@ private void LexSyntax()
         {
             while (More())
             {
-                LexDirective();
-
-                if (More())
+                if (IsDirective())
```

Comment:
This pattern is a little strange. Why not let the `LexDirective` method decide, as is the case in the rest of the code?